File: primatis-api/src/main/java/be/primatis/catalogue/dto/CreateCopyRequest.java
package be.primatis.catalogue.dto;

import be.primatis.catalogue.AvailabilityStatus;
import be.primatis.catalogue.CopyCondition;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;

/**
 * Contrat REST de création staff d'un {@code Copy} (DEV-06.6,
 * {@code COPY_MANAGE}) sous
 * {@code POST /api/v1/staff/titles/{titleId}/copies}. {@code titleId} vient
 * exclusivement du path, jamais du body. Aucun défaut métier caché :
 * {@code copyCondition}/{@code availabilityStatus} sont tous deux
 * obligatoires — le Service valide leur combinaison (LOST/OUT_OF_SERVICE
 * exigent UNAVAILABLE) et refuse {@code ON_LOAN}/{@code RESERVED} à la
 * création (exclusivement gérés par les futurs workflows Loan/Reservation,
 * DEV-07/08).
 */
public record CreateCopyRequest(
        @NotBlank String inventoryCode,
        String location,
        @NotNull CopyCondition copyCondition,
        @NotNull AvailabilityStatus availabilityStatus) {
}

File: primatis-web/src/app/catalogue/models/create-copy-request.ts
import { AvailabilityStatus } from './availability-status';
import { CopyCondition } from './copy-condition';

/**
 * Voir `be.primatis.catalogue.dto.CreateCopyRequest` côté backend, sous
 * `POST /api/v1/staff/titles/{titleId}/copies`. `titleId` vient
 * exclusivement du path — volontairement absent du body. Aucun défaut
 * métier côté frontend : `copyCondition`/`availabilityStatus` sont tous
 * deux obligatoires, le backend reste l'autorité de leur combinaison
 * (LOST/OUT_OF_SERVICE → UNAVAILABLE, ON_LOAN/RESERVED refusés à la
 * création).
 */
export interface CreateCopyRequest {
  inventoryCode: string;
  location?: string;
  copyCondition: CopyCondition;
  availabilityStatus: AvailabilityStatus;
}
